adaptdl/adaptdl/speedup.py

```diff
@@ -229,31 +257,36 @@ def fit(nodes, replicas, local_bsz, step_time, step_time_compute):
         params[5] = upper[5] = lower[5]
     if not any(replicas > 2):
         # Fix beta_n and beta_r if no replicas > 2.
-        params[3] = upper[3] = lower[3]
-        params[5] = upper[5] = lower[5]
+        params[3] = upper[3] = lower[5]
+        params[5] = upper[5] = lower[3]
     bounds = scipy.optimize.Bounds(lower, upper, keep_feasible=True)
-
-    args = (nodes, replicas, local_bsz, step_time, step_time_compute)
+    args = (nodes, replicas, local_bsz, grad_acc_steps,
+            step_time, step_time_compute, grad_acc_time)
     # FIXME: need to handle optimization failures and propagate to the Trainer.
     grad_fn = autograd.grad(_obj_fn)
     result = scipy.optimize.minimize(_obj_fn, params, args=args,
                                      jac=grad_fn, bounds=bounds)
     params = result.x
-
     np = orig_np  # Restore original numpy.
     return Params(*params)
 
 
-def _predict_log(params, nodes, replicas, local_bsz):
+def _predict_log(params, nodes, replicas, local_bsz, grad_acc_steps):
     params = Params(*params)
     step_time_compute = _predict_compute(params, local_bsz)
     step_time_network = _predict_network(params, nodes, replicas)
+    step_time_accumulation = step_time_compute
```

Comment:
Should this be `step_time_compute * grad_acc_steps`?

Reply:
`step_time_accumulation` is normalized by `grad_acc_steps`, to avoid grad(sqrt(x)) evaluated at zero when `grad_acc_steps` is zero.

Reply:
oh, I misread which part this is in

Reply:
the `step_time` computation will need `step_time_accumulation * grad_acc_steps`, but leaving `step_time_accumulation` to be as it is now

Reply:
Fixed